bpf/bpf_overlay.c

```diff
@@ -323,6 +329,49 @@ int tail_handle_ipv4(struct __ctx_buff *ctx)
 					      CTX_ACT_DROP, METRIC_INGRESS);
 	return ret;
 }
+
+#ifdef ENABLE_VTEP
+/*
+ * ARP responder for ARP requests from VTEP
+ * Respond to remote VTEP endpoint with cilium_vxlan MAC
+ */
+__section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_ARP)
+int tail_handle_arp(struct __ctx_buff *ctx)
```

Comment:
There is already a section in the `CILIUM_MAP_CALLS` for the `CILIUM_CALL_ARP` index. Doesn't this conflict with the other one?

Reply:
I have not thought about that, but I see `CILIUM_CALL_IPV4_FROM_LXC` also appeared in multiple places, something I am missing?

```
   6    639  bpf/bpf_host.c <<__section_tail>>
             __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV4_FROM_LXC)
  11    295  bpf/bpf_overlay.c <<__section_tail>>
             __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV4_FROM_LXC)
  13    117  bpf/bpf_xdp.c <<__section_tail>>
             __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV4_FROM_LXC)
```

Reply:
Ah, ok, if I have this right, the `CILIUM_MAP_CALLS` gets defined differently for each BPF object. So depending on whether the `bpf_host`, `bpf_overlay` or `bpf_xdp` object's program calls into the `CILIUM_CALL_IPV4_FROM_LXC` program (defined by these `__section_tail()` annotations), it should call into the version defined in the same file.

(This is because the actual name of the map is defined by `CALLS_MAP` in `bpf/lib/maps.h`, unlike `CILIUM_MAP_POLICY` which globally tracks the BPF program pointers for all endpoints on the node)

So the last snippet in the patch is actually what's calling into this code, the `ep_tail_call(ctx, CILIUM_CALL_ARP);`. :+1: